parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReader.java

```diff
@@ -29,5 +29,5 @@
 
   List<TripleIterator<?>> columns();
 
-  void setPageSource(PageReadStore pageStore);
+  void setPageSource(PageReadStore pageStore, long rowPosition);
```

Comment:
How about adding a default implementation that calls the old method? That way older implementations are still supported.

```java
  default void setPageSource(PageReadStore pageStore, long rowPosition) {
    setPageSource(pageStore);
  }

  default setPageSource(PageReadStore pageStore) {
    throw new UnsupportedOperationException("At least one setPageSource method must be implemented");
  }
```

Reply:
Also, then we can remove the changes from many of the implementations and cut down on the changes in this PR.